app/assets/javascripts/govuk_publishing_components/components/show-password.js

```diff
@@ -65,5 +66,15 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     this.input.setAttribute('type', 'password')
   }
 
+  ShowPassword.prototype.moveDataAttributesToButton = function () {
+    for (var i = 0; i < this.input.attributes.length; i++) {
```

Comment:
I wonder if we can compress this a little bit, feels quite verbose with the repetition of `this.input.attributes`. Could we store that as a variable that we can reference instead?


Additional idea I just had: how about instead of passing attributes for the button in a `data-button-somethingcustom="something"` format, we try using a single `data-button` attribute containing as many `name:value` pairs as needed. So what I'm saying is instead of: 
```
        button_do_this: do_this
        button_do_that: do_that
```
would it be possible to do something like:

```
  data_button:
        do_this: do_this
        do_that: do_that
```

Then instead of looping over all the attributes on the input element, identifying those which start with `data-button-` and processing them accordingly, the script could directly do `this.input. getAttribute('data-button')`, then parse the value which would be an object (probably, haven't actually checked)...

Hopefully that makes some sense, let me know what you think. 😅

Reply:
Good spot on the bug. A weird thing was happening - every time I removed an attribute, the list of attributes would shrink, so iterating to the next one would actually be the next one plus 1. The solution is a bit more verbose, but it works.

I liked the idea of putting the button attributes together but I'm not sure it solves a problem - it's just a different way of doing it. Either way I end up implementing a mechanism for handling two sets of data attributes (one for the input, one for the button) in Ruby or JS or both.